### PR TITLE
TypeScript Contained Language 2

### PR DESCRIPTION
**Pull request by jramsay**
A few changes to enable editing JavaScript blocks using TypeScript and Roslyn.

Creating a new pull request as the last one (https://github.com/dotnet/roslyn/pull/3354) had a commit that was not attributed correctly to my username.

Based on @jasonmalinowski, @Pilchie  CR comments I've removed the .html -> .html.js extension mapping out of Roslyn and instead we now handle it from the TypeScript language service.  

Adding other reviewers that were also on the previous pull request:
 @balajikris, @basoundr, @brettfo, @davkean, @DustinCampbell, @dpoeschl, @CyrusNajmabadi

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
:+1:. The content type stuff still seems a bit strange, but I forget exactly what that code was even originally added for. I suspect @balajikris or @basoundr might remember better than I.

**Comment by balajikris** [comment]
Hmm. I don't know why that was introduced, but it looks like we needed to determine the host type to conditionally support Rename and Formatting differently in Razor/Venus scenarios. @basoundr might know.

File: src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs [hunk 1]
```diff
@@ -113,8 +114,13 @@ private bool ShouldConsiderSymbol(ISymbol symbol)
...
         {
-            return document.Project.LanguageServices.GetService<IReferenceHighlightingAdditionalReferenceProvider>()
-                .GetAdditionalReferencesAsync(document, symbol, cancellationToken);
+            var additionalReferenceProvider = document.Project.LanguageServices.GetService<IReferenceHighlightingAdditionalReferenceProvider>();
+            if (additionalReferenceProvider != null)
+            {
+                return additionalReferenceProvider.GetAdditionalReferencesAsync(document, symbol, cancellationToken);
+            }
+
+            return Task.FromResult<IEnumerable<Location>>(Array.Empty<Location>());
```
Comment (balajikris) [inline src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs]: use SpecializedCollections.EmptyEnumerable for better allocation and perf?

**Comment by balajikris** [comment]
:+1:

**Comment by basoundr** [comment]
@jasonmalinowski @balajikris Just by looking at the usage, since we have special calculation for indentation within the C#/VB blocks in Razor, we need to know what kind of content we are working with.

File: src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs (hunk 1 above)
Comment (jramsay) [inline src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs]: thanks - done

**Comment by basoundr** [comment]
:+1:

**Pull request merged**